cmake_mtime_crc.py

```diff
@@ -0,0 +1,71 @@
+import os
+from time import time
+from zlib import crc32
+
+# #
+# Functions
+# #
+
+# get the CRC of a file's contents
+def getCRC32(file):
+    src_file = open(file, "r", encoding="utf-8")
+    file_contents = src_file.read()
+    src_file.close()
+
+    return crc32(file_contents.encode("utf8"))
+
+# recursively get all files in all subdirectories
+def getAllFiles(start):
+    file_list = []
+    
+    for root, directories, files in os.walk(start):
+        for file in files:
+            file_list.append(root+"/"+file)
+    
+    return file_list
+
+# #
+# Variables
+# #
+today = time()
+yesterday = time() - (24*60*60)
+mtime_file = "build/cmake_mtime_crc.txt"
+
+# #
+# Run
+# #
+file_list = getAllFiles("src")
+
+# get the current CRC all relevant files in src/, and set their mtime to yesterday
+crc_dict_curr = {}
+for file in file_list:
+    os.utime(file, (yesterday, yesterday))
+    if (file.endswith(".cpp") or file.endswith(".hpp") or file.endswith(".tpp") or file.endswith(".c") or file.endswith(".h")) and "CMakeFiles" not in file:
+        crc_dict_curr.update({file : getCRC32(file)})
+
+# if there's an existing stored list of CRCs, read that in
+if os.path.isfile(mtime_file):
+    readfile = open(mtime_file, "r", encoding="utf-8")
+    crc_dict_prev = {}
+    
+    for line in readfile:
+        line_list = line.strip().split(":")
+        crc_dict_prev.update({line_list[0] : line_list[1]})
```

Comment:
You now need to convert the CRCs back to integers on reading: `int(line_list[1])`

Reply:
So instead of converting it to a string initially, I convert it to a string at the end and convert something else to an integer in the mean time?

Reply:
Keep them as integers in memory.

When you save them to the file, convert them to strings (you're [already doing it](https://github.com/wesnoth/wesnoth/pull/2470/commits/2cc0a2d989653724ef4c3d6dece7a93cc50cfd1d#diff-81d567937bb911c42fb1742b689933b6R68)).

And when you load them from the file next time (this line), convert them back to integers.